xcell/cls/cov.py

```diff
@@ -112,16 +112,16 @@ def get_covariance_workspace(self):
 
     def _get_cl_for_cov(self, clab, clab_fid, ma, mb):
         mean_mamb = np.mean(ma * mb)
-        nl_cp = clab.get_ell_nl_cp()[1]
         if not mean_mamb:
-            cl_cp = np.zeros_like(nl_cp)
+            cl_cp = np.zeros((clab.get_n_cls(), 3*clab.nside))
         else:
             if isinstance(clab_fid, ClFid):  # Compute from theory
                 w = clab.get_workspace()
                 cl_cp = w.couple_cell(clab_fid.get_ell_cl()[1])
+                cl_cp += clab.get_ell_nl_cp()[1]
             else:  # Compute from data
```

Comment:
Could you add a comment here reminding me that at this stage, if it hasn't failed, it's because you have requested in the config file to fallback to the data Cell?

Reply:
Done